test/transactions/transactionBytes.js

```diff
@@ -12,53 +12,549 @@
+			const transactionBytes = getTransactionBytes(inTransferTransction);
+
+			(transactionBytes).should.be.eql(expectedBuffer);
+		});
+	});
+
+	describe('outTransfer transaction, type 7', () => {
+		const outTransferTransaction = {
+			type: 7,
+			amount: defaultAmount,
+			fee: 1 * fixedPoint,
+			recipientId: defaultRecipient,
+			senderPublicKey: defaultSenderPublicKey,
+			timestamp: defaultTimestamp,
+			asset: { outTransfer: { dappId: defaultAppId, transactionId: defaultTransactionId } },
+			signature: defaultSignature,
+			id: defaultTransactionId,
+		};
+		it('should return Buffer of type 7 (dapp outTransfer) transaction', () => {
+			const expectedBuffer = Buffer.from('B6opAgBdA2qFjOifhESRdi64niv71QpKCg2mWOSyYoslsReuCQDOvKqNNBU96AMAAAAAAAAxMjM0MjEzMTM5ODczNDg0MjA5MTMxMzg0MjJhilSXUhLq2T34yIFlXGJVRLzo7XzN/m8IpC7s+xrevQUTB75QFLsFFhe694FdUPYhKecJGBkDYeXU3UeWVBsK', 'base64');
+			const transactionBytes = getTransactionBytes(outTransferTransaction);
+
+			(transactionBytes).should.be.eql(expectedBuffer);
+		});
+	});
+});
+
+describe('getTransactionBytes helper functions', () => {
+	describe('#checkRequiredFields', () => {
+		const arrayToCheck = ['OneValue', 'SecondValue', 'ThirdValue'];
+		it('should accept array and object to check for required fields', () => {
+			const objectParameter = {
+				OneValue: '1',
+				SecondValue: '2',
+				ThirdValue: '3',
+			};
+
+			(checkRequiredFields(arrayToCheck, objectParameter)).should.be.true();
+		});
+
+		it('should throw on missing value', () => {
+			const objectParameter = {
+				OneValue: '1',
+				SecondValue: '2',
+			};
+
+			(checkRequiredFields.bind(null, arrayToCheck, objectParameter)).should.throw('ThirdValue is a required parameter.');
+		});
+	});
+
+	const defaultEmptyBuffer = Buffer.alloc(0);
+	describe('#getAssetDataForSendTransaction', () => {
+		it('should return Buffer for data asset', () => {
+			const expectedBuffer = Buffer.from('my data input', 'utf8');
+			const assetDataBuffer = getAssetDataForSendTransaction({
+				data: 'my data input',
+			});
+
+			(assetDataBuffer).should.be.eql(expectedBuffer);
+		});
+
+		it('should return empty Buffer for no asset data', () => {
+			const assetDataBuffer = getAssetDataForSendTransaction({});
+			(assetDataBuffer).should.be.eql(defaultEmptyBuffer);
+		});
+	});
+
+	describe('#getAssetDataForSignatureTransaction', () => {
+		it('should return Buffer for signature asset', () => {
+			const expectedBuffer = Buffer.from(defaultSenderPublicKey, 'hex');
+			const assetSignaturesPublicKeyBuffer = getAssetDataForSignatureTransaction({
+				signature: {
+					publicKey: defaultSenderPublicKey,
+				},
+			});
+
+			(assetSignaturesPublicKeyBuffer).should.be.eql(expectedBuffer);
+		});
+
+		it('should throw on missing publicKey in the signature asset', () => {
+			(getAssetDataForSignatureTransaction.bind(null, { signature: {} })).should.throw(
+				'publicKey is a required parameter.',
+			);
+		});
```

Comment:
As above: what if the key is present, but the value is `undefined`?

Reply:
Addressed in utils - fn isValidValue